src/java/org/apache/cassandra/db/SSTableImporter.java

```diff
@@ -97,7 +99,19 @@ synchronized List<String> importNewSSTables(Options options)
                         try
                         {
                             abortIfDraining();
-                            verifySSTableForImport(descriptor, entry.getValue(), options.verifyTokens, options.verifySSTables, options.extendedVerify);
+
+                            if (options.verifySSTables || options.verifyTokens)
+                                verifySSTableForImport(descriptor, entry.getValue(), options.verifyTokens, options.verifySSTables, options.extendedVerify);
+
+                            if (options.failOnMissingIndex)
```

Comment:
I put it here to be still in the "verification" stage of the import so we can fail fast before any sstables are actually imported.